scripts/check-docs/src/main.rs

```diff
@@ -0,0 +1,82 @@
+use std::process::Stdio;
+
+fn main() {
+    let rg_project = std::process::Command::new("rg")
+        .arg("--case-sensitive")
+        .arg("-g")
+        .arg("!/scripts/check-docs/src/main.rs")
+        .arg("ANCHOR")
+        .stdout(Stdio::piped())
+        .spawn()
+        .expect("failed rg command");
+
+    let output_files = std::process::Command::new("sed")
+        .arg("s/ //g; s=//==g")
+        .stdin(rg_project.stdout.unwrap())
+        .output()
+        .expect("failed sed command");
+
+    let mut stack = vec![];
+
+    let output_to_string =
+        String::from_utf8(output_files.stdout).expect("failed to parse command output");
+    let split = output_to_string.split('\n');
+    let vec_of_anchors = split
+        .into_iter()
+        .filter(|s| !s.is_empty())
+        .collect::<Vec<&str>>();
+
+    for i in vec_of_anchors {
+        let parts = i.split(':').collect::<Vec<&str>>();
```

Comment:
This can be simplified considerably if you use regexes to match and extract anchor data.

Reply:
I think regex is not a part of std lib.

Reply:
Why does it need to be? A requirement somewhere prohibiting dependencies?

Reply:
I was guided by this - https://github.com/FuelLabs/fuels-rs/pull/532#discussion_r949829467.

Reply:
Cargo will handle rust dependencies for you. [Regex](https://docs.rs/regex/latest/regex/) is a rust crate.  Therefore cargo will make sure you have it.

It will not scan the code to look for all binaries you've called and proceed to call your package manager to install them. That is why 'rg' was not an ideal choice.

Reply:
We'll talk.